querydsl-jpa/src/test/java/com/querydsl/jpa/AbstractJPATest.java

```diff
@@ -346,6 +402,18 @@ public void Case5() {
         assertEquals(ImmutableList.of(0, 1, 1, 1),
                 query().from(cat).orderBy(cat.id.asc())
                        .select(cat.mate.when(savedCats.get(0)).then(0).otherwise(1)).fetch());
+
+        List<Integer> rv = query().from(cat)
+            .select(Expressions.cases()
+                    .when(cat.toes.in(2, 3)).then(cat.id.multiply(cat.toes))
+                    .otherwise(4)).fetch();
+        assertInstancesOf(Integer.class, rv);
```

Comment:
This duplicates Case3 right? It [came from there](https://github.com/querydsl/querydsl/commit/9f5b019c223e502ac10f92e22c1b08c1e4d5a160#diff-7d971dd1c9376907a88a32f2d2c5881cR370).
I think that you could move this assertion there, the projection is already there.